src/Equinox.Core/Category.fs

```diff
@@ -32,42 +32,38 @@ open System.Threading.Tasks
 /// Store-agnostic baseline functionality for a Category of 'event representations that fold to a given 'state
 [<NoComparison; NoEquality>]
 type Category<'event, 'state, 'context>
-    (   resolveInner: string -> string-> struct (Core.ICategory<'event, 'state, 'context> * string * (CancellationToken -> Task<unit>) voption),
+    (   name,
+        resolveStream: string -> string -> struct (Core.ICategory<'event, 'state, 'context> * string * (CancellationToken -> Task<unit>) voption),
         empty: struct (Core.StreamToken * 'state)) =
     /// Provides access to the low level store operations used for Loading and/or Syncing updates via the Decider
-    /// (Normal usage is via the adjacent `module Decider` / `Factory.Resolve` helpers)
-    member _.Stream(log: Serilog.ILogger, context: 'context, categoryName, streamId) =
-        let struct (inner, streamName, init) = resolveInner categoryName streamId
+    /// (Normal usage is via the adjacent `module Decider` / `Stream.Resolve` helpers)
+    member _.Stream(log: Serilog.ILogger, context: 'context, streamId) =
+        let struct (inner, streamName, init) = resolveStream name streamId
```

Comment:
Now's your chance to drop `init` too 😈

Reply:
reasoning: it's cosmos specific and amounts to the cosmos category passing itself a value through the core category

Reply:
I have a failure of imagination as to how to achieve that though..

The objective is for the first append per container in any process to do the `CreateStoreProcIfNotExists` (unless you opt out, which would mean you'd need to guarantee to run an `eqx init` has happened for every container every time the stored proc code changes/is versioned)

Also, if there are concurrent first appends to multiple streams (across multiple containers), the inits happen exactly once per container 

(aside from latency, a null stored proc updated costs 6 RU I believe)

Reply:
> (unless you opt out, which would mean you'd need to guarantee to run an eqx init has happened for every container every time the stored proc code changes/is versioned)

I assumed this was the recommended way?

Reply:
that said, it sounds like you want an async lazy somewhere?

Reply:
https://github.com/jet/equinox/commit/bd8a771eff4eb0e71bfec9b09544401a2f438faf spiked it, it works, let's do it 💪